sdk/src/main/java/io/radar/sdk/RadarBeaconUtils.kt

```diff
@@ -0,0 +1,70 @@
+package io.radar.sdk
+
+import android.bluetooth.le.ScanFilter
+import android.bluetooth.le.ScanRecord
+import android.os.Build
+import android.util.Log
+import androidx.annotation.RequiresApi
+import io.radar.sdk.model.RadarBeacon
+import java.nio.ByteBuffer
+import java.util.*
+
+@RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+internal object RadarBeaconUtils {
+
+    private const val MANUFACTURER_ID = 76
+
+    fun getScanFilter(beacon: RadarBeacon): ScanFilter? {
+        val uuid = UUID.fromString(beacon.uuid)
+        val major = beacon.major.toInt()
+        val minor = beacon.minor.toInt()
+
+        val manufacturerData = ByteBuffer.allocate(23)
```

Comment:
It's nice to have comment with a link to where blocks of code like this are taken from so people can read more about it (unless it was coded from scratch).

Reply:
Yep, I'll add a bunch of comments in this class.